MovieApp/MovieCategorisedCollectionView.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  MovieCategorisedCollectionView.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 12.04.2023..
+//
+
+import Foundation
+import UIKit
+import PureLayout
+import MovieAppData
+
+class MovieCategorisedCollectionView : UIView {
+    
+    var categoryCollectionView : UICollectionView!
+    var categoryLabel : UILabel!
+    var movieList : [MovieModel]!
+    var categoryString : String!
+    let reuseId = "cell"
+    
+    init(category: String, listOfMovies: [MovieModel]) {
+        super.init(frame: .zero)
+        categoryString = category
+        movieList = listOfMovies
+        buildViews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        categoryString = ""
+        movieList = []
+        buildViews()
+    }
```

Comment:
You could do it like this. And remove force unwraps on `categoryString` and `movieList`.

```suggestion
    init(category: String, listOfMovies: [MovieModel]) {
        self.categoryString = category
        self.movieList = listOfMovies
        buildViews()
        super.init(nibName: nil, bundle: nil)
    }
    
   required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
```